src/Umbraco.Core/Collections/ObservableDictionary.cs

```diff
@@ -235,15 +236,76 @@ public bool TryGetValue(TKey key, out TValue val)
 
         #endregion
 
+        /// <summary>
+        /// The exception that is thrown when a duplicate key inserted.
+        /// </summary>
+        /// <seealso cref="System.Collections.ObjectModel.ObservableCollection{TValue}" />
+        /// <seealso cref="System.Collections.Generic.IReadOnlyDictionary{TKey, TValue}" />
+        /// <seealso cref="System.Collections.Generic.IDictionary{TKey, TValue}" />
+        [Serializable]
         internal class DuplicateKeyException : Exception
```

Comment:
All exceptions should be `public`, as consumers can't catch it otherwise!

Reply:
This exception also seems redundant, as it's only used in its parent class (`ObservableDictionary<TKey, TValue>`) and can be replaced with a `ArgumentException` (like https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.add).

Reply:
The `InvalidOperationException` thrown by `ObservableDictionary<TKey, TValue>` when the key isn't found should also be changed to a `KeyNotFoundException` (like https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.item).